scripts/msg.h.template

```diff
@@ -183,26 +184,26 @@ namespace message_traits
 {
 
 @{
-bool_traits = dict(IsMessage=True,
-                   IsFixedSize=gencpp.is_fixed_length(spec, msg_context, search_path),
-                   HasHeader=spec.has_header(),
-                   )
+bool_traits = OrderedDict(
+    IsMessage=True,
+    IsFixedSize=gencpp.is_fixed_length(spec, msg_context, search_path),
+    HasHeader=spec.has_header(),
+)
 def booltotype(b):
    return "TrueType" if b else "FalseType"
 }
 
 // BOOLTRAITS @bool_traits
-// @search_path
 
 @# TODO
-// !!!!!!!!!!! @(dir(spec))
+// !!!!!!!!!!! @(sorted(dir(spec)))
 @#if spec.is_fixed_length():
 @#    traits = traits.append('IsFixedSize')
 
 @#template <class ContainerAllocator>
-@#struct IsFixedSize<@(cpp_full_name_with_alloc) >:: @(dir(spec)) ? "TrueType" ! "FalseType") { };
+@#struct IsFixedSize<@(cpp_full_name_with_alloc) >:: @(sorted(dir(spec))) ? "TrueType" ! "FalseType") { };
 @#template <class ContainerAllocator>
-@#struct IsFixedSize<@(cpp_full_name_with_alloc) const >:: @(dir(spec)) ? "TrueType" ! "FalseType") { };
+@#struct IsFixedSize<@(cpp_full_name_with_alloc) const >:: @(sorted(dir(spec))) ? "TrueType" ! "FalseType") { };
```

Comment:
All of these are only template comments which are never part of the generated header. Reading through the comments it seems all of this belongs under the `TODO` from line 197/198 but has been implemented already below.

How about deleting all lines starting from `// BOOLTRAITS @bool_traits` until this line? Neither seems to contains any useful information in the generated header.

Reply:
I haven't looked into how empy works, but at least the `IsFixedSize` seem to generate method definitions in the generated header. I would be fine with deleting from BOOLTRAITS to !!!!!.

Reply:
> I haven't looked into how empy works

Anything behind `@#` is a comment and won't appear in the expanded template.

> but at least the `IsFixedSize` seem to generate method definitions in the generated header.

The corresponding template snippet is in the for loop below.

Reply:
Done, thanks for the heads up :).